docs: reference Grafana Agent Flow as an OTel Collector distribution

tpaschalis opened the pull request:
This PR adds documentation references to Grafana Agent Flow's role as an OpenTelemetry collector distribution in the main landing page as well as the root of the Flow mode docs.

tpaschalis commented on docs/sources/_index.md:
```diff
@@ -15,10 +15,9 @@ cascade:
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
+inspired by [Terraform][]. It is designed to be flexible, performant, and
+compatible with multiple ecosystems such as Prometheus and OpenTelemetry.
```
I'm OK with the first suggestion. 

Let me look if there's a good place for in the root README for the second snippet as well.

cyrille-leclerc commented on docs/sources/_index.md:
```diff
@@ -15,10 +15,9 @@ cascade:
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector with configuration
```
cc @jpkrohling

tpaschalis commented:
Upon Nth reading, I don't like the repeated `capabilities` wording here, but will take a fresh look a little later on.

cyrille-leclerc commented on docs/sources/_index.md:
```diff
@@ -15,10 +15,9 @@ cascade:
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
+inspired by [Terraform][]. It is designed to be flexible, performant, and
```
Should we call out as primary goal brining Prometheus native pipeline to he OTel Collector in order to notable seamless integration between the Otel and Prometheus worlds?

cyrille-leclerc commented on docs/sources/_index.md.t:
```diff
@@ -15,10 +15,9 @@ cascade:
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector with configuration
```
cc @jpkrohling

cyrille-leclerc commented on docs/sources/flow/_index.md:
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
 * Use expressions to bind components together to build a programmable pipeline.
 * Includes a UI for debugging the state of a pipeline.
 
+{{< param "PRODUCT_NAME" >}} is a [distribution][] of the OpenTelemetry
```
Ditto above "is Grafana's distribution of the OpenTelemetry Collector"

cyrille-leclerc commented on docs/sources/flow/_index.md:
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
+Each distribution offers a different collection of components and capabilities.
+{{< param "PRODUCT_NAME" >}} is a distribution that includes dozens of
+OpenTelemetry-native [components][] from the OpenTelemetry project and
+introduces new capabilities such as programmable pipelines, clustering support,
```
```suggestion
introduces new capabilities such as native Prometheus pipelines, code defined pipelines, clustering support,
```

cyrille-leclerc reviewed (commented):
Can we review the phrasing "Grafana Agent is an OpenTelemetry Collector distribution", @jpkrohling confirmed recently it's more framed as "Grafana's distribution of the OpenTelemetry Collector".

Moreover, I feel the primary value added to call out is "native Prometheus pipelines"

clayton-cornell commented on README.md:
```diff
@@ -1,9 +1,8 @@
-configuration inspired by [Terraform][]. It is designed to be flexible,
-performant, and compatible with multiple ecosystems such as Prometheus and
-OpenTelemetry.
+Grafana Agent is an OpenTelemetry Collector distribution with configuration
```
```suggestion
Grafana Agent is Grafana's distribution of the OpenTelemetry Collector distribution with configuration
```

tpaschalis replied:
After discussing offline, we decided to go with the original verbiage for now.

clayton-cornell commented on docs/sources/flow/_index.md:
```diff
@@ -33,6 +33,17 @@ Components allow for reusability, composability, and focus on a single task.
 
+{{< param "PRODUCT_NAME" >}} is a [distribution][] of the OpenTelemetry
+Collector by Grafana Labs.
+Each distribution offers a different collection of components and capabilities.
+{{< param "PRODUCT_NAME" >}} is a distribution that includes dozens of
```
```suggestion
{{< param "PRODUCT_NAME" >}} includes dozens of
```

Is it necessary to define what the distributions do?

tpaschalis replied:
Not sure, we were on the fence but it might be worth to explain what makes a/our distribution different.

clayton-cornell commented on docs/sources/flow/_index.md:
```diff
@@ -94,5 +105,8 @@ This feature is experimental, and it doesn't support all River components.
+[Reference]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/reference/"
+[components]: "/docs/agent/ -> /docs/agent/<AGENT_VERSION>/flow/reference/"
+[components]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/send-data/agent/flow/reference/components"
+[distribution]: "https://opentelemetry.io/ecosystem/distributions/"
```
Move this link definition out of the `docs/reference` block (and when you do, remove the double quotes around the URL). The link definition can go above or below this block... I usually put them above.

clayton-cornell reviewed (commented):
Overall, the changes look good. I've added a couple of comments/suggestions

tpaschalis commented:
cc @clayton-cornell I've split the paragraph flow/_index.md into smaller chunks and replaced some duplicate words, let me know if it sounds better now

jpkrohling commented:
I agree with @cyrille-leclerc that it should be phrased as

> Grafana Agent is Grafana's distribution of the OpenTelemetry Collector

clayton-cornell commented on docs/sources/flow/_index.md:
```diff
@@ -33,6 +33,19 @@ Components allow for reusability, composability, and focus on a single task.
+
+In addition to being an OpenTelemetry Collector distribution,
+{{< param "PRODUCT_NAME" >}} also includes first-class support for the
+Prometheus and Loki ecosystems, enabling pipelines to mix-and-match as needed.
```
```suggestion
Prometheus and Loki ecosystems, allowing you to mix-and-match your pipelines.
```

clayton-cornell reviewed (approved):
One more doc change.

The text is consistent enough and in-line with what we've been discussing. I think it'll be OK as it is. It seems to meet the requirements we've discussed.
